athena-storage-api/src/main/java/com/amazonaws/athena/storage/datasource/CsvDatasource.java

```diff
@@ -0,0 +1,342 @@
+    {
+        checkFilesSize(bucketName, objectNames);
+        return getStorageSplits(bucketName, objectNames);
+    }
+
+    /**
+     * Returns splits, usually by page size with offset and limit so that lambda can parallelize to load data against a given SQL statement
+     *
+     * @param bucketName Name of the bucket
+     * @param fileName   Name of the file under the bucket
+     * @return An instance of {@link StorageSplit}
+     */
+    @Override
+    public List<StorageSplit> getStorageSplits(final String bucketName,
+                                               final String fileName)
+    {
+        GcsOfflineStream offlineGcsStream = FileCacheFactory.createOfflineGcsStream(storage, bucketName, fileName);
+        requireNonNull(offlineGcsStream, "Stream cached was null for file " + fileName + " under bucket " + bucketName);
+        long totalRecords = StorageUtil.getCsvRecordCount(offlineGcsStream.file());
+        return GcsCsvSplitUtil.getStorageSplitList(totalRecords, fileName, recordsPerSplit());
+    }
+
+    /**
+     * Retrieves table data for provided arguments
+     *
+     * @param schema      Schema of the table
+     * @param constraints Constraints if any
+     * @param tableInfo   Table info containing table and schema name
+     * @param split       Current Split instance
+     */
+    @Override
+    public void readRecords(Schema schema, Constraints constraints, TableName tableInfo,
+                            Split split, BlockSpiller spiller, QueryStatusChecker queryStatusChecker)
+    {
+        String databaseName = tableInfo.getSchemaName();
+        String tableName = tableInfo.getTableName();
+        if (!storeCheckingComplete) {
+            this.checkMetastoreForAll(databaseName);
+        }
+        String bucketName;
+        List<String> objectNames = null;
+        bucketName = databaseBuckets.get(databaseName);
+        if (bucketName == null) {
+            throw new DatabaseNotFoundException("No schema '" + databaseName + "' found");
+        }
+        Map<String, List<String>> tableObjectMap = tableObjects.get(databaseName);
+        if (tableObjectMap != null) {
+            objectNames = tableObjectMap.get(tableName);
+        }
+        if (objectNames == null) {
+            throw new TableNotFoundException("No table '" + tableName + "' found under schema '" + databaseName + "'");
+        }
+        try {
+            readRecordsForRequest(schema, constraints, tableInfo, split, storage, bucketName, spiller, queryStatusChecker);
+        }
+        catch (Exception exception) {
+            throw new UncheckedStorageDatasourceException("Error occurred during reading CSV records. Table name : " + tableName
+                    + ", schema name: " + tableInfo.getSchemaName() + ", original file name(s): " + objectNames
+                    + ", bucket name: " +  bucketName + ". Error message: "
+                    + exception.getMessage(), exception);
+        }
```

Comment:
There's no need to try catch here just to enrich the exception message.
Instead, you can just log all the information you want prior to making the call to `readRecordsForRequest()`.

Reply:
try/catch block has been removed